src/controllers/auth/forgotPassword.ts

```diff
@@ -12,11 +13,29 @@ export const postForgot = async (req: Request, res: Response): Promise<Response>
   if (!errors.isEmpty()) {
     return res.status(422).json({ errors: errors.array() })
   }
-  const email = req.body.email
-  const token = createNewToken()
-  // TODO: Save bcrypt hashedToken within database passwordResets table
-  // const payload = getPayload(token)
-  // can logger.info(token) for now, otherwise we want to email this to the user.
-  logger.info(token)
-  return res.status(501).json({ email, token })
+
+  let user: User
+
+  try {
+    const email = req.body.email
+    user = await User.findOneOrFail({ where: { email } })
+  } catch (error) {
+    logger.error(error)
+    return res.status(404).json({ error: 'email not found' })
+  }
+
+  await PasswordReset.delete(user.id)
+  const token = await createNewToken()
+  const passwordReset = await createNewPasswordReset(user, token)
+
+  try {
+    await PasswordReset.save(passwordReset)
+  } catch (error) {
+    logger.error(error)
+    return res.status(500).json({ error: 'something went wrong' })
+  }
+
+  // TODO: email the token to the user/guest, return a 204 response
+  logger.info(passwordReset.token)
```

Comment:
You can update this to log `token`:
```suggestion
   logger.info(token) 
```
Since this is the correct value to pass to the guest reset password endpoint.

Reply:
Fixed